### PR TITLE
refactor!: orb tools 12 migration

### PR DESCRIPTION
**Pull request by brivu**
## Description

This `PR` updates the `AWS SAM Severless` Orb to the new major version of Orb Tools. As it contains breaking changes, several modifications were unnecessary to move from v11 to [v12](https://circleci.com/developer/orbs/orb/circleci/orb-tools). 

## Changes

1. All parameters and commands containing dashes have been converted to the new snake case standard (replace `-` with `_`).
2. The comprehensive list of changes can be found below.

### Changes in `.circleci/config.yml`:

1. Update the orb-tools version from 11.6 to 12.0.
1. Move the job requirement list from `orb-tools/publish` to `orb-tools/continue`.
1. Remove the `orb-tools/publish` job since development versions are no longer necessary.
1. Add the new `orb_name` parameter in `orb-tools/continue`.
1. Rename the `orb-tools/continue` job parameters to comply with the new snake case standard.
   - `orb_name`, `pipeline_number` and `vcs_type`.

### Changes in `.circleci/test-deploy.yml`:

1. Update the orb-tools version from `11.6` to `12.0`.
1. Remove the `aws-cli: circleci/aws-cli@dev:<<pipeline.git.revision>>` line, and replace it with `aws-cli: {}`.
1. Remove the `orb-tools/lint`, `orb-tools/pack`, and `orb-tools/review` jobs. ⚠️ 
1. Rename the `orb-tools/publish` job parameters to comply with the new snake case standard.
   - `orb_name`, `vcs_type`, `pub_type`, `enable_pr_comment` and `github_token`.
1. Change the `orb-tools/pack` filter to trigger only on tagged releases.

⚠️: This change is not related to the migration itself. These jobs were redundant in `test-deploy.yml` as they already run in `config.yml`. See [Orb-Template #2](https://github.com/CircleCI-Public/Orb-Template/pull/2/files).

### CODE REVIEW

File: src/commands/deploy.yml [hunk 1]
```diff
@@ -5,65 +5,64 @@ parameters:
...
-  s3-bucket:
+  s3_bucket:
     type: string
     description: The name of the S3 bucket where this command uploads the artifacts that are referenced in your template. If this option is set, the SAM package command will be run automatically.
     default: ""
-  resolve-s3:
+  resolve_s3:
     type: boolean
-    description: Automatically create an Amazon S3 bucket to use for packaging and deploying for non-guided deployments. If you specify the --guided option, then the AWS SAM CLI ignores --resolve-s3. If you specify both the --s3-bucket and --resolve-s3 options, then an error occurs.
+    description: Automatically create an Amazon S3 bucket to use for packaging and deploying for non-guided deployments. If you specify the --guided option, then the AWS SAM CLI ignores --resolve_s3. If you specify both the --s3_bucket and --resolve_s3 options, then an error occurs.
```
Comment (EricRibeiro) [inline src/commands/deploy.yml]: If we are referring to the CLI's parameters then it should be:

```suggestion
    description: Automatically create an Amazon S3 bucket to use for packaging and deploying for non-guided deployments. If you specify the --guided option, then the AWS SAM CLI ignores --resolve-s3. If you specify both the --s3-bucket and --resolve-s3 options, then an error occurs.
```

File: src/commands/install.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ parameters:
     type: string
     default: latest
   python_version:
-    description: 'If set, this version of Python will be installed and set with pyenv globally. ex: "3.7.0" This is only for the local environment and will not have any effect if use-container is enabled.'
+    description: 'If set, this version of Python will be installed and set with pyenv globally. ex: "3.7.0" This is only for the local environment and will not have any effect if use_container is enabled.'
```
Comment (EricRibeiro) [inline src/commands/install.yml:9]: Is `use-container` an orb-related thing? If not, then it should remain unchanged.

File: src/examples/build_test_deploy_oidc.yml [hunk 3]
```diff
@@ -21,12 +21,12 @@ usage:
...
                   profile: OIDC-USER
-                  role-arn: "arn:aws:iam::123456789012:role/VALID_OIDC_SAM_ROLE"
+                  role_arn: "arn:aws:iam::123456789012:role/VALID_OIDC_SAM_ROLE"
             # Must use same profile configured in aws-cli/setup command
             profile: OIDC-USER
             template: ./path/to/template.yml
-            stack-name: staging-stack
-            s3-bucket: my-s3-bucket
+            stack_name: staging-stack
+            s3_bucket: my-s3_bucket
```
Comment (EricRibeiro) [inline src/examples/build_test_deploy_oidc.yml]:
```suggestion
            s3_bucket: my-s3-bucket
```

File: src/executors/default.yml [hunk 4]
```diff
@@ -1,10 +1,10 @@
 description: >
   Machine executor provided by CircleCI. Needed for privileged access to Docker for the SAM CLI.
 parameters:
-  machine-image:
+  machine_image:
     default: ubuntu-2204:2022.07.1
     type: string
     description: >
-      The CircleCI Linux Machine VM Image for this job. Find other available machine images here: https://circleci.com/docs/2.0/configuration-reference/#available-machine-images
+      The CircleCI Linux Machine VM Image for this job. Find other available machine images here: https://circleci.com/docs/2.0/configuration-reference/#available-machine_images
```
Comment (EricRibeiro) [inline src/executors/default.yml]:
```suggestion
      The CircleCI Linux Machine VM Image for this job. Find other available machine images here: https://circleci.com/docs/2.0/configuration-reference/#available-machine-images
```

File: src/jobs/deploy.yml [hunk 5]
```diff
@@ -2,22 +2,22 @@ description: >
...
     description: >
-      The CircleCI Linux Machine VM Image for this job. Find other available machine images here: https://circleci.com/docs/2.0/configuration-reference/#available-machine-images
+      The CircleCI Linux Machine VM Image for this job. Find other available machine images here: https://circleci.com/docs/2.0/configuration-reference/#available-machine_images
   version:
     description: 'SAM CLI version to be installed.'
     type: string
     default: latest
   python_version:
-    description: 'If set, this version of Python will be installed and set with pyenv globally. ex: "3.7.0" This is only for the local environment and will not have any effect if use-container is enabled.9'
+    description: 'If set, this version of Python will be installed and set with pyenv globally. ex: "3.7.0" This is only for the local environment and will not have any effect if use_container is enabled.9'
```
Comment (EricRibeiro) [inline src/jobs/deploy.yml]:
```suggestion
    description: 'If set, this version of Python will be installed and set with pyenv globally. ex: "3.7.0" This is only for the local environment and will not have any effect if use-container is enabled.'
```

Comment (EricRibeiro) [reply src/jobs/deploy.yml]: If it's really supposed to be `use_container` ignore the suggestion but remember to remove that `9` at the end.

File: src/scripts/build.sh [hunk 6]
```diff
@@ -1,25 +1,27 @@
...
+if [ -n "$ORB_EVAL_BUILD_DIR" ]; then
+  set -- "$@" --build-dir "${ORB_EVAL_BUILD_DIR}"
 fi
-if [ -n "$SAM_PARAM_BASE_DIR" ]; then
-  set -- "$@" --base-dir "$(eval "echo $SAM_PARAM_BASE_DIR")"
+if [ -n "$ORB_EVAL_BASE_DIR" ]; then
+  set -- "$@" --base-dir "${ORB_EVAL_BASE_DIR}"
 fi
-if [ "$SAM_PARAM_CONTAINER" = 1 ]; then
+if [ "$ORB_BOOL_USE_CONTAINER" = 1 ]; then
```
Comment (EricRibeiro) [inline src/scripts/build.sh]: Consider using `-eq` if the left-side value is a number. See https://stackoverflow.com/questions/18668556/how-can-i-compare-numbers-in-bash

**Review by EricRibeiro — CHANGES_REQUESTED** [review]

File: src/commands/install.yml (hunk 2 above)
Comment (brivu) [reply src/commands/install.yml:9]: That's talking about the actual `use_container` parameter. It's a boolean value that defaults to `true` so it's an orb related thing

File: src/jobs/deploy.yml (hunk 5 above)
Comment (brivu) [reply src/jobs/deploy.yml]: It's supposed to be `use_container`. I removed the `9`. Thank you!

File: src/scripts/build.sh (hunk 6 above)
Comment (brivu) [reply src/scripts/build.sh]: Good tip but the left side in this case, `ORB_BOOL_USE_CONTAINER` is a boolean value.

Comment (brivu) [reply src/scripts/build.sh]: I changed the `=` to `-eq`.

**Review by EricRibeiro — APPROVED** [review]

**Pull request merged**